tests/test_deserialization.py

```diff
@@ -0,0 +1,264 @@
+from datadog_api_client.v1.model.synthetics_test_request import SyntheticsTestRequest
+from datadog_api_client.v1.model_utils import validate_and_convert_types
+from datadog_api_client.v1 import Configuration as Configuration
+from datadog_api_client.v2.model.logs_archive import LogsArchive
+from datadog_api_client.v2.model.logs_archive_destination import LogsArchiveDestination
+from datadog_api_client.v2.model_utils import validate_and_convert_types as validate_and_convert_types_v2
+from datadog_api_client.v2 import Configuration as ConfigurationV2
+
+
+def test_unknown_nested_oneof_in_list():
+    body = """{
+        "status": "paused",
+        "public_id": "jv7-wfd-kvt",
+        "tags": [],
+        "locations": [
+            "pl:pl-kevin-y-6382df0d72d4588e1817f090b131541f"
+        ],
+        "message": "",
+        "name": "Test on www.example.com",
+        "monitor_id": 28558768,
+        "type": "api",
+        "created_at": "2021-01-12T10:11:40.802074+00:00",
+        "modified_at": "2021-01-22T16:42:10.520384+00:00",
+        "subtype": "http",
+        "config": {
+            "request": {
+                "url": "https://www.example.com",
+                "method": "GET",
+                "timeout": 30
+            },
+            "assertions": [
+                {
+                    "operator": "lessThan",
+                    "type": "responseTime",
+                    "target": 1000
+                },
+                {
+                    "operator": "is",
+                    "type": "statusCode",
+                    "target": 200
+                },
+                {
+                    "operator": "A non existent operator",
+                    "type": "body",
+                    "target": {
+                        "xPath": "//html/head/title",
+                        "operator": "contains",
+                        "targetValue": "Example"
+                    }
+                }
+            ],
+            "configVariables": []
+        },
+        "options": {
+            "monitor_options": {
+                "notify_audit": false,
+                "locked": false,
+                "include_tags": true,
+                "new_host_delay": 300,
+                "notify_no_data": false,
+                "renotify_interval": 0
+            },
+            "retry": {
+                "count": 0,
+                "interval": 300
+            },
+            "min_location_failed": 1,
+            "min_failure_duration": 0,
+            "tick_every": 60
+        }
+    }"""
+    config = Configuration()
+    deserialized_data = validate_and_convert_types(
+        json.loads(body), (SyntheticsAPITest,), ["received_data"], True, True, config)
+    assert isinstance(deserialized_data, SyntheticsAPITest)
+    assert len(deserialized_data.config.assertions) == 3
+    assert deserialized_data.config.assertions[2].operator == "A non existent operator"
```

Comment:
Can you add checks for the `_unparsed` attribute, to make sure it has the value we expect ?

Reply:
Done!